--- /beetsplug/bpmanalyser/command.py
+++ /beetsplug/bpmanalyser/command.py
@@ -14,6 +14,8 @@
 from beets.library import Library as BeatsLibrary
 from beets.ui import Subcommand, decargs
 
+from beetsplug.bpmanalyser.get_song_bpm import _analyse_tempo
+
 # Module methods
 log = logging.getLogger('beets.bpmanalyser')
 
@@ -161,22 +163,8 @@ def analyse(item):
         self.execute_on_items(items, analyse, msg='Analysing tempo...')
 
     def get_bpm_from_analyser_script(self, item_path):
-        log.debug(
-            "calling external script: {}".format(self.analyser_script_path))
-
-        proc = Popen([sys.executable, self.analyser_script_path, item_path],
-                     stdout=PIPE, stderr=PIPE)
-        stdout, stderr = proc.communicate()
-
-        try:
-            bpm = int(stdout.decode("utf-8"))
-            errors = ""
-        except ValueError:
-            bpm = 0
-            errors = stderr.decode()
-            if len(errors) > 0:
-                log.debug(errors)
-
+        bpm = _analyse_tempo(item_path)
+        errors = ""
         return bpm, errors
 
     def execute_on_items(self, items, func, msg=None):
